address_test.go

```diff
@@ -1,71 +1,34 @@
 package iotago_test
 
 import (
-	"errors"
-	"github.com/iotaledger/hive.go/serializer"
-	"github.com/iotaledger/iota.go/v2/tpkg"
 	"testing"
 
-	"github.com/iotaledger/iota.go/v2"
+	"github.com/iotaledger/iota.go/v3"
+	"github.com/iotaledger/iota.go/v3/tpkg"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestEd25519Address_Deserialize(t *testing.T) {
-	tests := []struct {
-		name       string
-		edAddrData []byte
-		err        error
-	}{
+func TestAddressDeSerialize(t *testing.T) {
+	tests := []deSerializeTest{
 		{
-			"ok",
-			func() []byte {
-				_, edAddrData := tpkg.RandEd25519Address()
-				return edAddrData
-			}(),
-			nil,
+			name:   "ok - Ed25519Address",
+			source: tpkg.RandEd25519Address(),
+			target: &iotago.Ed25519Address{},
 		},
 		{
-			"not enough bytes",
-			func() []byte {
-				_, edAddrData := tpkg.RandEd25519Address()
-				return edAddrData[:iotago.Ed25519AddressSerializedBytesSize-1]
-			}(),
-			serializer.ErrDeserializationNotEnoughData,
+			name:   "ok - AliasAddress",
+			source: tpkg.RandAliasAddress(),
+			target: &iotago.AliasAddress{},
 		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			edAddr := &iotago.Ed25519Address{}
-			bytesRead, err := edAddr.Deserialize(tt.edAddrData, serializer.DeSeriModePerformValidation)
-			if tt.err != nil {
-				assert.True(t, errors.Is(err, tt.err))
-				return
-			}
-			assert.NoError(t, err)
-			assert.Equal(t, len(tt.edAddrData), bytesRead)
-			assert.Equal(t, tt.edAddrData[serializer.SmallTypeDenotationByteSize:], edAddr[:])
-		})
-	}
-}
-
-func TestEd25519Address_Serialize(t *testing.T) {
-	originEdAddr, originData := tpkg.RandEd25519Address()
-	tests := []struct {
-		name   string
-		source *iotago.Ed25519Address
-		target []byte
-	}{
 		{
-			"ok", originEdAddr, originData,
+			name:   "ok - NFTAddress",
+			source: tpkg.RandEd25519Address(),
```

Comment:
This should be a NFT address.

Reply:
a6e066c9f3503b4f804aedd8bedf059dfc542855